components/org.wso2.carbon.identity.application.authenticator.hypr/src/main/java/org/wso2/carbon/identity/application/authenticator/hypr/web/HYPRAuthorizationAPIClient.java

```diff
@@ -0,0 +1,157 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authenticator.hypr.web;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.http.HttpResponse;
+import org.wso2.carbon.identity.application.authenticator.hypr.HyprAuthenticatorConstants;
+import org.wso2.carbon.identity.application.authenticator.hypr.exception.HYPRAuthnFailedException;
+
+import java.io.IOException;
+import java.security.NoSuchAlgorithmException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * The HYPRAuthorizationAPIClient class contains all the functions related to handling the API calls to the HYPR server.
+ **/
+public class HYPRAuthorizationAPIClient {
+
+    private final ObjectMapper objectMapper;
+    // HYPR Configuration parameters
+    private final String baseUrl;
+    private final String appId;
+    private final String apiToken;
+
+    public HYPRAuthorizationAPIClient(final String baseUrl, final String appId, final String apiToken) {
+        this.baseUrl = baseUrl;
+        this.appId = appId;
+        this.apiToken = apiToken;
+        this.objectMapper = new ObjectMapper();
+        this.objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+    }
+
+    /**
+     * Call the HYPR server API to retrieve the registered devices.
+     *
+     * @param username The username provided by the user.
+     * @return response         A HTTPResponse object.
+     * @throws HYPRAuthnFailedException Exception throws when there is an error occurred when retrieving the
+     *                                  registered devices via the api call
+     */
+    public HttpResponse getRegisteredDevicesRequest(String username) throws HYPRAuthnFailedException {
+        //{{baseUrl}}/rp/ api/oob/client/authentication/{{appId}}/{{username}}/devices
+        // TODO : Look into a way to use the URLBuilder
+        String deviceInfoURL = String.format("%s%s%s/%s/devices", this.baseUrl,
+                HyprAuthenticatorConstants.HYPR.HYPR_USER_DEVICE_INFO_PATH, this.appId, username);
+
+        try {
+            HttpResponse response = HYPRWebUtils.jsonGet(this.apiToken, deviceInfoURL);
+
+            if (response.getStatusLine().getStatusCode() == 401) {
+                throw new HYPRAuthnFailedException(
+                        HyprAuthenticatorConstants.ErrorMessages.
+                                HYPR_ENDPOINT_API_TOKEN_INVALID_FAILURE.getCode(),
+                        HyprAuthenticatorConstants.ErrorMessages.
+                                HYPR_ENDPOINT_API_TOKEN_INVALID_FAILURE.getMessage());
+            } else if (response.getStatusLine().getStatusCode() > 400) {
```

Comment:
```suggestion
            } else if (response.getStatusLine().getStatusCode() >= 400) {
```
WDYT?

Reply:
The corresponding HYPR API  doesn't give 400 error code at any circumstance.